### ISSUE TITLE
muselsl via BLE112 not streaming on Windows 10

### ISSUE THREAD

**Issue by mgpritchard** [alexandrebarachant/muse-lsl#68]
Hiya, I'm running on python3.7, 32bit Win10 on a pretty old laptop that doesn't have a built-in bluetooth receiver so am using a BLE112 dongle and the bgapi backend.

muselsl will happily find the device but when I start a stream it's timing out almost immediately, looks like its just not getting any data through:

` $ muselsl stream -b bgapi --address 00:55:DA:B3:9A:2C `
` c:\users\michael\appdata\local\programs\python\python37-32\lib\site-packages\sklearn\externals\joblib\externals\cloudpickle\cloudpickle.py:47: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses `
`  import imp `
`Connecting to Muse : 00:55:DA:B3:9A:2C...`
`Connected.`
`Streaming...`
`Disconnected. `

Seems like a very similar issue to #59 
Any advice?

EDIT: managed to get a visualiser to try and view the stream before it disconnects. Showing up as empty traces from -5s to 0s which further implies that it's not receiving samples at all
![eegtrace_nostream](https://user-images.githubusercontent.com/45095451/48619006-ff265780-e992-11e8-8ee7-f1bfbf4588ab.png)
muselsl view then reports:
`Stream transmission broke off (Input stream error.); re-connecting...`

**Comment by jdpigeon** [comment]
This is really interesting, and makes me think that muse-lsl's current dependencies might not be working with the bgapi backend.

Take a look at this thread and the corresponding fork and dependencies file that @amandakeasson recommends: https://github.com/NeuroTechX/eeg-notebooks/issues/3

If that helps get data streaming on your computer, please post back here and we can update the dependencies for everyone!

**Comment by mgpritchard** [comment]
Thanks for the advice, that does seem to have done the trick!
Running now in a miniconda environment with python 3.6.7, and the dependencies from https://github.com/amandakeasson/eeg-notebooks/blob/master/requirements_mac.txt

Now getting data through! The stream does report:
`missing sample 0 : 65535`
but the stream no longer kicks out and is definitely receiving data:
![image](https://user-images.githubusercontent.com/45095451/48733053-59e0dd00-ec39-11e8-9ce2-caf2021ae9f6.png)


For completeness I should mention that I've also implemented the `time.sleep(.25)` fix referenced in peplin/pygatt/issues/118. I'd already implemented that on Friday when the stream still wasn't working so it seems to be a wholly separate issue to do with pygatt accessing the COM port, but worth mentioning.

**Comment by mgpritchard** [comment]
No worries, couldn't have done it without your suggestion
If I had the free time I'd go through and try to establish which individual dependencies were specifically at fault... Something for a rainy day perhaps

**Comment by rschmaelzle** [comment]
hello, i am not sure if my error is related. I am - after running amandakeasson's mac-specific adjustments - able to run the notebooks to connect the muse. then, in a separate terminal, i am also able to run muselsl view, check the signals, and finally run sample experiments. 
however, I now tried to make everything more automatic, and particularly I tried to run the code to connect the muse in a simple python file (code  see below - or I can  run 'muselsl stream' alone from the commandline). However, whenever I leave the jupyter notebooks, the muse does initially connect ("streaming..."), but then after about 0.5 sec it says "Disconnected". Any ideas why it runs smoothly in the notebooks but not in the terminal or via a python script (e.g. python3 connect_muse.py)?

Best, r

KEy code that is in the jupyter notebook or the python file:
`
from muselsl import stream, list_muses, view, record
from multiprocessing import Process
muses = list_muses()

stream_process = Process(target=stream, args=(muses[0]['address'],))
stream_process.start()`
